Sources/Version.swift

```diff
@@ -1,39 +1,131 @@
-import ObjectMapper
 
-struct VersionTransform: TransformType {
+public enum Version: Codable {
+    indirect case subversion(UInt, Version)
+    case version(UInt)
 
-    typealias Object = Version
-    typealias JSON = String
+    fileprivate var value: UInt {
+        switch self {
+        case .subversion(let version, _):
+            return version
+        case .version(let version):
+            return version
+        }
+    }
 
-    func transformFromJSON(_ value: Any?) -> Version? {
-        var convertible: LosslessStringConvertible? = value as? String
-        convertible = convertible ?? value.flatMap({ $0 as? Double })
-        convertible = convertible ?? value.flatMap({ $0 as? Float })
-        guard let string = convertible?.description else {
+    private var subversion: Version? {
+        switch self {
+        case .subversion(_, let subversion):
+            return subversion
+        case .version:
             return nil
         }
+    }
+
+    public var major: UInt { return self.value }
+    public var minor: UInt? { return self.subversion?.value }
+    public var patch: UInt? { return self.subversion?.subversion?.value }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
Should we add a convenience initializer that takes a `String` (that this initializer calls)? I imagine that could be useful for version comparisons by the consumer.

Reply:
Sure